File: doc-examples/nitrite-example-java/src/main/java/example/Widget.java
package example;

import io.micronaut.data.annotation.GeneratedValue;
import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.MappedEntity;

import java.util.UUID;

// tag::uuid-widget[]
@MappedEntity("widgets")
public class Widget {
    @Id
    @GeneratedValue
    private UUID id;

    private String name;

    public Widget() {
    }

    public Widget(String name) {
        this.name = name;
    }

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
}
// end::uuid-widget[]

File: doc-examples/nitrite-example-kotlin/src/main/kotlin/example/Widget.kt
package example

import io.micronaut.data.annotation.GeneratedValue
import io.micronaut.data.annotation.Id
import io.micronaut.data.annotation.MappedEntity
import java.util.UUID

// tag::uuid-widget[]
@MappedEntity("widgets")
class Widget {
    @Id
    @GeneratedValue
    var id: UUID? = null

    var name: String? = null
    // end::uuid-widget[]

    constructor()

    // tag::uuid-widget[]
    constructor(name: String) {
        this.name = name
    }
    // end::uuid-widget[]
// tag::uuid-widget[]
}
// end::uuid-widget[]
